mesonbuild/modules/gnome.py

```diff
@@ -1016,55 +1017,71 @@ def yelp(self, state: 'ModuleState', args: T.Tuple[str, T.List[str]], kwargs) ->
-
-        poargs = state.environment.get_build_command() + [
-            '--internal', 'yelphelper', 'update-po',
-            '--subdir=' + state.subdir,
-            '--id=' + project_id,
-            '--sources=' + source_str,
-            '--langs=' + '@@'.join(langs),
-        ]
-        potarget = build.RunTarget('help-' + project_id + '-update-po', poargs,
-                                   [], state.subdir, state.subproject)
-
-        rv = [inscript, pottarget, potarget]
-        return ModuleReturnValue(None, rv)
+        pot_file = os.path.join('@SOURCE_ROOT@', state.subdir, 'C', project_id + '.pot')
+        pot_args = ['itstool', '-o', pot_file] + [os.path.join('@SOURCE_ROOT@', state.subdir, 'C', s) for s in sources]
+        pottarget = build.RunTarget(f'help-{project_id}-pot', pot_args, [],
+                                    os.path.join(state.subdir, 'C'), state.subproject)
+        targets.append(pottarget)
+
+        install_dir = kwargs.get('install_dir', os.path.join(state.environment.get_datadir(), 'help'))
+        for l in langs:
+            l_subdir = os.path.join(state.subdir, l)
+            po_file = l + '.po'
+
+            po_args = ['msgmerge', '-q', '-o', os.path.join('@SOURCE_ROOT@', l_subdir, po_file),
+                       os.path.join('@SOURCE_ROOT@', l_subdir, po_file), pot_file]
+            potarget = build.RunTarget(f'help-{project_id}-{l}-update-po',
+                                       po_args, [pottarget], l_subdir, state.subproject)
+            targets.append(potarget)
+            potargets.append(potarget)
+
+            gmo_kwargs = {'command': ['msgfmt', '@INPUT@', '-o', '@OUTPUT@'],
+                          'input': po_file,
+                          'output': l + '.gmo',
+            }
+            gmotarget = build.CustomTarget('', l_subdir, state.subproject, gmo_kwargs)
+            targets.append(gmotarget)
+
+            merge_kwargs = {'command': ['itstool', '-m', os.path.join(l_subdir, l + '.gmo'),
+                                        '-o', '@OUTDIR@', '@INPUT@'],
+                            'input': sources_files,
+                            'output': sources,
+                            'depends': gmotarget,
+                            'install': True,
+                            'install_dir': os.path.join(install_dir, l, project_id),
+            }
+            mergetarget = build.CustomTarget('', l_subdir, state.subproject, merge_kwargs)
+            targets.append(mergetarget)
+
+        allpotarget = build.AliasTarget(f'help-{project_id}-update-po', potargets,
+                                        state.subdir, state.subproject)
+        targets.append(allpotarget)
+
+        c_data = self.interpreter.install_data_impl(sources_files, os.path.join(install_dir, 'C', project_id),
+                                                    mesonlib.FileMode(), [], [])
+        targets.append(c_data)
```

Comment:
I guess I have misunderstood something from your comment, but this is not working (neither if `c_data` is not added to the targets). Any clue of what might I be doing wrong? I've spent a fair amount of time trying to understand why is this not installing the data, but still no clue..,

Reply:
Sorry, I meant to take a look at how to use a `build.Data()` -- you cannot invoke interpreter functions directly from a module, it's complicated.

This works:

```
-        c_data = self.interpreter.install_data_impl(sources_files, os.path.join(install_dir, 'C', project_id),
-                                                    mesonlib.FileMode(), [], [])
+        final_installdir = os.path.join(install_dir, 'C', project_id)
+        c_data = build.Data(sources_files, final_installdir, final_installdir ,mesonlib.FileMode(), state.subproject)
```

Reply:
Oh, sorry my confusion, that works perfectly!